crates/autopilot/src/domain/competition/winner_selection/combinatorial.rs

```diff
@@ -350,3 +350,748 @@ type ScoreByDirection = HashMap<DirectedTokenPair, Score>;
+        }
+    }
+
+    fn address(id: u64) -> H160 {
+        H160::from_low_u64_le(id)
+    }
+
+    fn create_order(
+        uid: u64,
+        sell_token: H160,
+        sell_amount: eth::U256,
+        buy_token: H160,
+        buy_amount: eth::U256,
+    ) -> Order {
+        let mock_protocol_fees = vec![fee::Policy::Surplus {
+            factor: fee::FeeFactor::try_from(0.0).unwrap(),
+            max_volume_factor: fee::FeeFactor::try_from(0.0).unwrap(),
+        }];
+
+        // build the UID of the order
+        let mut encoded_uid = [0u8; 56];
+        let uid_bytes = uid.to_le_bytes();
+        encoded_uid[..uid_bytes.len()].copy_from_slice(&uid_bytes);
+        let encoded_uid = OrderUid(encoded_uid);
+
+        Order {
+            uid: encoded_uid,
+            sell: eth::Asset {
+                amount: sell_amount.into(),
+                token: sell_token.into(),
+            },
+            buy: eth::Asset {
+                amount: buy_amount.into(),
+                token: buy_token.into(),
+            },
+            protocol_fees: mock_protocol_fees,
+            side: order::Side::Sell,
+            receiver: Some(H160::zero().into()),
+            owner: H160::zero().into(),
+            partially_fillable: false,
+            executed: eth::U256::zero().into(),
+            pre_interactions: vec![],
+            post_interactions: vec![],
+            sell_token_balance: order::SellTokenSource::Erc20,
+            buy_token_balance: order::BuyTokenDestination::Erc20,
+            app_data: AppDataHash(hex!(
+                "6000000000000000000000000000000000000000000000000000000000000007"
+            )),
+            created: Default::default(),
+            valid_to: Default::default(),
+            signature: order::Signature::PreSign,
+            quote: None,
+        }
+    }
+
+    fn create_price(value: eth::U256) -> Price {
+        Price::try_new(eth::Ether(value)).unwrap()
+    }
+
+    fn create_auction(
+        orders: Vec<Order>,
+        prices: Option<HashMap<eth::TokenAddress, Price>>,
+    ) -> Auction {
+        // Initialize the prices of the tokens if they are not specified
+        let prices = prices.unwrap_or({
+            let default_price = create_price(DEFAULT_TOKEN_PRICE.into());
+            let mut res = HashMap::new();
+            for order in &orders {
+                res.insert(order.buy.token, default_price);
+                res.insert(order.sell.token, default_price);
+            }
+            res
+        });
+
+        Auction {
+            id: 0,
+            block: 0,
+            orders,
+            prices,
+            surplus_capturing_jit_order_owners: vec![],
+        }
+    }
+
+    fn create_trade(
+        order: &Order,
+        executed_sell: eth::U256,
+        executed_buy: eth::U256,
+    ) -> TradedOrder {
+        TradedOrder {
+            side: order::Side::Sell,
```

Comment:
This assumes all orders are sell orders.

Reply:
Updated the JSON test cases so they specify the side (`"buy"` or `"sell"`), and created a unit tests specifically for buy orders.

Reply:
But wouldn't we still need a `match` here that sets this to buy or sell depending on the order that gets settled?